### ISSUE TITLE
Empty response body when failing to authenticate with client credentials

### ISSUE THREAD

**Issue by amy-langley** [zooniverse/panoptes#2257]
We had a problem where caesar was configured with invalid credentials. It was throwing a rather cryptic error: `JSON::ParserError: A JSON text must at least contain two octets!` 

It turns out that this was because requesting `oauth/token` with invalid credentials is returning an empty response body with an HTTP 422 error, but @zwolf said that it should return a JSON object that includes an error message.

I'm going to try to modify the faraday panoptes middleware to behave more sensibly (in this case, fail more legibly), but I'm entering this issue in case there is a problem with the oauth stuff we're doing that is resulting in this behavior. I've attached a VCR fixture that shows the failing request.

**Comment by amyrebecca** [comment]
[should_make_a_get_request.yml.txt](https://github.com/zooniverse/Panoptes/files/859530/should_make_a_get_request.yml.txt)

**Comment by amyrebecca** [comment]
Related to zooniverse/faraday-panoptes#1

**Comment by zwolf** [comment]
I said that it should return a JSON object that includes an error message. Even if we technically should be returning a error object with the 422, the status code is probably enough to go on.

**Comment by amyrebecca** [comment]
Sorry, yeah, that's more sensible. As you can see in the referenced PR I'm now checking for an empty response body or a 422 and throwing an actually meaningful exception so that it can either be a) caught and handled by client code or b) will at least show up in the logs as something we can understand.

**Comment by amyrebecca** [comment]
I am wondering if someone thought `''.to_json` seemed redundant and replaced it with `''` because that is an easy mistake to make and would also produce the error we're seeing

**Comment by camallen** [comment]
I thinks this is related to the custom security of the doorkeeper app [here](https://github.com/zooniverse/Panoptes/blob/master/app/controllers/tokens_controller.rb#L16) the head response would explain the busted body json parsing.

**Comment by zwolf** [comment]
Yeah, I wasn't clear on whether that was applying the header/code or if it was rendering it, but it's def tossing those back with nothing included. Rendering a spec-compliant message instead's a good idea.

**Comment by amyrebecca** [comment]
I did put a workaround in our client middleware if it turns out this is too much of a hassle to deal with. It's mostly a diagnostic issue the next time something is misconfigured :)

**Comment by camallen** [comment]
Decent response body messaging on this would be super useful as y'all found out ;)
